builder_test.go

```diff
@@ -20,3 +20,33 @@ func TestBuilderTransparency(t *testing.T) {
 		require.NotEqual(t, testType, err.Type())
 	})
 }
+
+func TestBuilderRespectsNoStackTrace(t *testing.T) {
+	wrapperErrorTypes := []*Type{testTypeSilent, testTypeSilentTransparent}
+
+	for _, et := range wrapperErrorTypes {
+		t.Run(et.String(), func(t *testing.T) {
+			t.Run("Naked", func(t *testing.T) {
+				err := NewErrorBuilder(et).
+					WithCause(errors.New("naked error")).
+					Create()
+				require.Nil(t, err.stackTrace)
+			})
+
+			t.Run("WithoutStacktrace", func(t *testing.T) {
+				err := NewErrorBuilder(et).
+					WithCause(testTypeSilent.NewWithNoMessage()).
+					Create()
+				require.Nil(t, err.stackTrace)
+			})
+
+			t.Run("WithStacktrace", func(t *testing.T) {
+				err := NewErrorBuilder(et).
+					WithCause(testType.NewWithNoMessage()).
+					Create()
+				require.Nil(t, err.stackTrace)
```

Comment:
This particular case troubles me. I'm not entirely sure how the case of 'no-trace type is used to wrap whatever other error' is used. 
On the one hand, such types are mostly used without wrapping other errors - so, if you do wrap, maybe that's exactly why you added the wrapping type, and you only use it this way. And you expect to avoid dragging any nested stack traces along. 
On the other hand, such logic loses the original information. You already paid for its collection, and it may be a nasty surprise for user.

Maybe a better path would be not to enhance stack trace, but not to lose the existing one either?

Reply:
> On the other hand, such logic loses the original information.

We don't lose anything here. The stack trace in `cause` is still there. Give me a moment, I'll verify it in tests.

Reply:
It looks like I'm wrong and although technically we don't lose anything, `fmt.Sprintf("%+v", err)` is not going to print nested stack trace if `err` doesn't have it's own stack trace. It looks like a bug, IMO.

Reply:
OK, it may be my mistake in interpreting the test code. Thank you, explicitly testing that part as well would be perfect.

Reply:
> It looks like a bug, IMO.
It all depends on observable behaviour. Probably, until this point, this made sense.

Reply:
Okay, I'll use less offensive words then 🙂 
This behaviour doesn't seem correct taking into account the changes in the current PR. I should either fix my PR or fix the `(*Error).Format` method. Let me think a little bit...

Reply:
@PeterIvanov, I've added a new test to ensure that `ErrorBuilder` keeps stack trace from cause even if error type explicitly requests stack trace omission.

I had to update `testify` to get access to `Same` assertion method.

Reply:
I've got zero issues with your wording. My point was that the we should change that if it's needed now - but to consider other use cases possibly affected.

Reply:
@PeterIvanov, please, check out new version. I've tried to address the concerns you've highlighted.

Reply:
Sorry for the force-push, I had to do it due to rebase. However, I've embraced it as the opportunity to glue all my commits into one 🙂